File: week_1_c/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_bricks(int bricks);
void print_spaces(int spaces);
void print_row(int spaces, int bricks);


int main(void)
{
    //declare height variable
    int height = get_height();

    //for loop to print pyramid with spaces and bricks
    for(int i=0; i< height; i++)
        {
            print_row(height-(i+1), i+1);
        }
}

//get pyramid height from user
int get_height(void)
{
    int height;
    do
        {
            height = get_int("Height: ");
        }
    while(height<1);
    return height;
}

//print number of spaces
void print_spaces(int spaces)
{
    for(int i=0; i<spaces; i++)
        {
            printf(" ");
        }
}

//print number of bricks
void print_bricks(int bricks)
{
    for(int i=0; i<bricks; i++)
        {
            printf("#");
        }
}

//print row- print spaces + print bricks
void print_row(int spaces, int bricks)
{
    print_spaces(spaces);
    print_bricks(bricks);
    printf("\n");
}

File: week_6_python/sentimental-mario-more/mario.py
# import moduels
from cs50 import get_int


def main():
    pyramid = get_height()
    # create a for loop to print out space, blocks, gap, blocks
    # print_row(height-(i+1), i+1, 2*height+2-2*height);
    for i in range(pyramid):
        print_row(pyramid - (i + 1), i + 1, 2*pyramid+2-2*pyramid)


def get_height():
    # get user to input a height (number between 1 and 8 inclusive)
    while True:
        height = get_int("Pick a height between 1 and 8 ")
        if height >= 1 and height <= 8:  # check to make sure user input in between 1 & 8
            break
    return height


def print_spaces(spaces):
    for i in range(spaces):
        print(" ", end="")


def print_bricks(bricks):
    for i in range(bricks):
        print("#", end="")


def print_gap(gap):
    for i in range(gap):
        print(" ", end="")


def print_row(spaces, bricks, gap):
    print_spaces(spaces)
    print_bricks(bricks)
    print_gap(gap)
    print_bricks(bricks)
    print()


main()
